packages/client/src/open-feature.ts

```diff
@@ -37,7 +37,9 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI<Provider> implements Ma
   async setContext(context: EvaluationContext): Promise<void> {
     const oldContext = this._context;
     this._context = context;
-    await this._defaultProvider?.onContextChange?.(oldContext, context);
+
+    const allProviders = [this._defaultProvider, ...this._clientProviders.values()];
+    await Promise.all(allProviders.map((provider) => provider.onContextChange?.(oldContext, context)));
```

Comment:
Promise.all would fail fast if any of the `onContextChange` methods threw. I think we should catch and log an error or use Promise.allSettled.

Reply:
I agree. One provider's error should not impact others. We may need to mention this in the spec.

Reply:
Good point, will add it. That was actually my question in the issue.

Reply:
@beeme1mr @toddbaert Fixed it now